src/schrodinger/rdkit_extensions/helm/monomer_coordgen.cpp

```
@@ -59,9 +61,28 @@ const std::string POLYMER_ID{"polymerID"};
 // Replacement atom label for monomers with SMILES strings as their atom label
 const std::string SMILES_MONOMER_LABEL{"CX"};
 
-enum ChainDirection { LTR, RTL };
-enum BranchDirection { UP, DOWN };
-enum PolygonStartSide { LEFT, RIGHT };
+enum class ChainDirection { LTR, RTL };
+enum class BranchDirection { UP, DOWN };
+enum class Direction { N, S, E, W, NW, NE, SW, SE };
+enum class PolygonStartSide { LEFT, RIGHT };
+static const std::map<Direction, RDGeom::Point3D> DIRECTION_TO_POINT_MAP = {
+    {Direction::N, RDGeom::Point3D(0, 1, 0)},
+    {Direction::S, RDGeom::Point3D(0, -1, 0)},
+    {Direction::E, RDGeom::Point3D(1, 0, 0)},
+    {Direction::W, RDGeom::Point3D(-1, 0, 0)},
+    {Direction::NE, RDGeom::Point3D(1, 1, 0)},
+    {Direction::NW, RDGeom::Point3D(-1, 1, 0)},
+    {Direction::SE, RDGeom::Point3D(1, -1, 0)},
+    {Direction::SW, RDGeom::Point3D(-1, -1, 0)}};
+
+static RDGeom::Point3D direction_to_point(Direction dir)
+{
+    auto it = DIRECTION_TO_POINT_MAP.find(dir);
+    if (it != DIRECTION_TO_POINT_MAP.end()) {
+        return it->second;
+    }
+    return RDGeom::Point3D(0, 0, 0);
```

Based on the contents of `DIRECTIONS_TO_POINT_MAP`, it seems like we already account for all possible values of `DIRECTION`. If we end up getting a value that isn't accounted for (e.g., if we add a new `DIRECTION` value in the future), would we really want to silently ignore that, or would it make more sense to raise an exception?